File: src/com/codingbaby/ohmyidea/action/ui/ShowHelpDialogAction.java
package com.codingbaby.ohmyidea.action.ui;

import com.intellij.openapi.actionSystem.AnAction;
import com.intellij.openapi.actionSystem.AnActionEvent;
import com.intellij.openapi.project.DumbAware;

/**
 * Created by baby on 15/3/7.
 */

public class ShowHelpDialogAction extends AnAction implements DumbAware {

    private static HelpDialog helpDialog;

    @Override
    public void actionPerformed(AnActionEvent anActionEvent) {
        if (helpDialog != null && helpDialog.isVisible()) {
            helpDialog.dispose();
        }
        helpDialog = new HelpDialog();
        helpDialog.show();
    }


}

File: src/main/java/com/codingbaby/ohmyidea/action/ShowHelpDialogAction.kt
package com.codingbaby.ohmyidea.action

import com.codingbaby.ohmyidea.action.ui.HelpDialog
import com.intellij.openapi.actionSystem.AnActionEvent
import com.intellij.openapi.project.DumbAwareAction

/**
 * Created by baby on 15/3/7.
 */

class ShowHelpDialogAction : DumbAwareAction() {

    override fun actionPerformed(anActionEvent: AnActionEvent) {
        if (helpDialog != null && helpDialog!!.isVisible) {
            helpDialog!!.dispose()
        }
        helpDialog = HelpDialog()
        helpDialog!!.show()
    }

    companion object {

        private var helpDialog: HelpDialog? = null
    }


}